linode/manager
# upcoming: [M3-8019] – Add Encrypted/Not Encrypted status to Linode Detail summary header

## Description 📝
Add Encrypted/Not Encrypted status to Linode Detail summary header.

## Changes  🔄
- Add `lke_cluster_id` to Linode interface, update mocks and several test files accordingly
- Add `Encrypted` and `Not Encrypted` status to Linode Detail summary header

## Target release date 🗓️
6/10

## Preview 📷
<details>
<summary>Screenshots</summary>

Encrypted:
![Screenshot 2024-05-31 at 2 55 17 PM](https://github.com/linode/manager/assets/114682940/5daf62d8-b939-4a5c-98b1-51f9e131feae)

Unencrypted & is LKE Linode:
![Screenshot 2024-05-31 at 2 57 22 PM](https://github.com/linode/manager/assets/114682940/cd74097e-5415-4562-8342-bbd142716931)

Unencrypted & is standard Linode:
![Screenshot 2024-05-31 at 2 58 22 PM](https://github.com/linode/manager/assets/114682940/8d2d877d-d24f-4e2f-b57d-e3a60859a1d5)

</details>

## How to test 🧪
To test each of the three cases outlined by the screenshots, it'll be easiest to use the MSW and make adjustments to these lines:

https://github.com/linode/manager/blob/0e2d367d192f9a28b92b5b37ebd7f566c345ef5c/packages/manager/src/mocks/serverHandlers.ts#L768-L778

```typescript
      linodeFactory.build({
        backups: { enabled: false },
        disk_encryption: 'disabled', // change to 'enabled' to see encrypted state
        id,
        label: 'Gecko Edge Test',
        lke_cluster_id: null, // change to a number to see the LKE Linode state
        region: 'us-edge-1',
      })
```

### Prerequisites
1. Have the linode_disk_encryption tag on your account
2. Point at the alpha/dev environment
3. Have the Linode Disk Encryption (LDE) flag in our dev tool toggled on

### Verification steps
Without the tag and/or with the LDE flag in our dev tool toggled off, you should not see any LDE-related things in the UI. Otherwise,

1. Navigate to a Linode Details page
2. Confirm the displayed encryption status matches what the `disk_encryption` field in the linode detail network request indicates (`enabled` --> `Encrypted`, `disabled` --> `Not Encrypted`)
3. If "Not Encrypted", standard Linodes should have a tooltip that reads: "Use Rebuild to enable or disable disk encryption." LKE Linodes should have a tooltip that reads: "To enable disk encryption, delete the node pool and create a new node pool. New node pools are always encrypted." 
4. Ensure the header looks clean as you shrink and expand the viewport (made some styling adjustments to accommodate the encrypted status)

## As an Author I have considered 🤔
